internal/stationxml/v1.1/data_availability_span.go

```diff
@@ -0,0 +1,42 @@
+package stationxml
+
+import (
+	"fmt"
+)
+
+// NOTE: missing exta attributes via other namespace
+
+// A type for describing data availability spans, with variable
+// continuity. The time range described may be based on the request parameters that
+// generated the document and not necessarily relate to continuity outside of the
+// range. It may also be a smaller time window than the request depending on the data
+// characteristics.
+type DataAvailabilitySpan struct {
+	Start DateTime `xml:"start,attr"`
+	End   DateTime `xml:"end,attr"`
+
+	// The number of continuous time series segments contained in the
+	// specified time range. A value of 1 indicates that the time series is continuous
+	// from start to end.
+	NumberSegments int `xml:"numberSegments.attr"`
+
+	// The maximum time tear (gap or overlap) in seconds between time
+	// series segments in the specified range. </xs:documentation>
+	MaximumTimeTear *float64 `xml:"maximumTimeTear,attr,omitempty"`
+}
+
+func (d DataAvailabilitySpan) IsValid() error {
+
+	if v := d.NumberSegments; v < 1 {
+		return fmt.Errorf("invalid number of segments: %d", v)
```

Comment:
can be `d.NumberSegments < 1`?

Reply:
but then I don't have the v as a nice short variable for the error message (just an idiosyncrasy of mine, much like my avoidance of >= and <=).